lib/attime.rb

```diff
@@ -0,0 +1,61 @@
+# This helper is trying to replicate the relative parts of
+# https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/render/attime.py
+
+module CheckGraphite
+  DAY = 86400
+  UNITS = {
+    "s" => "second",
+    "se" => "second",
+    "sec" => "second",
+    "second" => 1,
+    "seconds" => "second",
+
+    "min" => "minute",
+    "minute" => 60,
+    "minutes" => "minute",
+
+    "ho" => "hour",
+    "hour" => 3600,
+    "hours" => "hour",
+
+    "d" => "day",
+    "da" => "day",
+    "day" => DAY,
+    "days" => "day",
+
+    "week" => 7 * DAY,
+    "weeks" => "week",
+
+    "m" => nil, # min or mon?
+    "mon" => "month",
+    "month" => 30 * DAY,
+    "months" => "month",
+
+    "y" => "year",
+    "ye" => "year",
+    "year" => 365 * DAY,
+    "years" => "year",
+  }
+  EXPR = /([+-])?([0-9]*)(.*)/
```

Comment:
The current implementation will yield "Bad unit s 2d 3y" which I feel is a good enough error message for this. `.*` at end is already anchored to end of input (unless there are newlines). However, the current implementation does not handle floats: `1.5days` => `Bad unit .5days` which are easier to support than fail intelligently. Fixed.